### ISSUE TITLE
Docs: `projectService` - consider adding an alternative to `import.meta.dirname`

### ISSUE THREAD

**Issue by AndreaPontrandolfo** [typescript-eslint/typescript-eslint#9781]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Right now the docs suggest adding this for typed-linting:

```js
tsconfigRootDir: import.meta.dirname
```

However, this syntax is only supported since Node `v20.11.0`, which, arguably, is a VERY new version of Node, which many people probably don't have installed in their systems.
Maybe consider adding a more retro-compatible syntax.
I hacked together this:

```js
import * as url from 'url';

export const getLegacyCompatDirname = () => {
  const isESM = typeof import.meta !== 'undefined';
  return url.fileURLToPath(new URL('.', isESM ? import.meta.url : __filename));
};
```

But i'm not 100% sure it's correct, and the bundler still complains with a warning too (although i think it's fine regardless).

### Affected URL(s)

https://typescript-eslint.io/getting-started/typed-linting

**Comment by AndreaPontrandolfo** [comment]
Oh, i just noticed that a link to a workaround is already there, i totally missed it.
Closing, sorry.
